subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/VersionInSettingsPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,241 @@
+                }
+            }
+"""
+        buildScript "plugins { id '$PLUGIN_ID' }"
+
+        then:
+        verifyPluginApplied('2.0')
+    }
+
+    def "can override plugin version in settings script"() {
+        when:
+        buildScript "plugins { id '$PLUGIN_ID' version '2.0' }"
+
+        then:
+        verifyPluginApplied('2.0')
+    }
+
+    def "can use plugin version from settings script in one of sibling projects"() {
+        when:
+        settingsFile << "include 'p1', 'p2'"
+
+        file("p1/build.gradle") << """
+            plugins { 
+                id '$PLUGIN_ID' 
+            }
+            ${verifyPluginTask('1.0')}
+"""
+        file("p2/build.gradle") << """
+            plugins { 
+                id '$PLUGIN_ID' version '2.0'
+            }
+            ${verifyPluginTask('2.0')}
+"""
+
+        then:
+        succeeds "verify"
+    }
+
+    def "ignores plugin version from settings script when plugin loaded in parent project"() {
+        when:
+        settingsFile << "include 'p1'"
+
+        buildFile << """
+            plugins { 
+                id '$PLUGIN_ID' version '2.0'
+            }
+            ${verifyPluginTask('2.0')}
+"""
+        file("p1/build.gradle") << """
+            plugins { 
+                id '$PLUGIN_ID'
+            }
+            ${verifyPluginTask('2.0')}
+"""
+
+        then:
+        succeeds "verify"
+    }
+
+    def "ignores plugin version from settings script when plugin added as buildscript dependency"() {
+        when:
+        buildFile << """
+            buildscript {
+                dependencies {
+                    classpath "my:plugin:2.0"
+                }
+            }
+            plugins { 
+                id '$PLUGIN_ID'
+            }
+"""
+
+        then:
+        verifyPluginApplied('2.0')
+    }
+
+    def "ignores plugin version from settings script when plugin added as buildSrc"() {
+        when:
+        file('buildSrc/build.gradle') << """
+            repositories {
+                gradlePluginPortal()
+            }
+            dependencies {
+                implementation "my:plugin:2.0"
+            }
+"""
+        buildFile << """
+            plugins { 
+                id '$PLUGIN_ID'
+            }
+"""
+
+        then:
+        verifyPluginApplied('2.0')
+    }
+
+    def "cannot request that plugin be applied in settings script"() {
+        when:
+        withSettings """
+            pluginManagement {
+                plugins {
+                    id '$PLUGIN_ID' version '1.0' apply true
+                }
+            }
+"""
+
+        then:
+        fails "help"
+    }
+
+    def "cannot specify plugin version twice in settings script"() {
+        when:
+        withSettings """
+            pluginManagement {
+                plugins {
+                    id '$PLUGIN_ID' version '1.0'
+                    id '$PLUGIN_ID' version '2.0'
+                }
+            }
+"""
+
+        then:
+        fails "help"
```

Comment:
Should we validate that the error message here is something sane and comprehensible?

Reply:
Yes, yes we should :)

Reply:
995ea7062ede5adaadb28730821aacc149c8257b